archive_converter.py

```diff
@@ -14,65 +14,71 @@
                 continue
             email = generate_email_from_file(os.path.join(data_dir, filename))
-            if email.id not in self._messages_seen:
+            if not self._message_dao.get(email.id):
                 new_messages.append(email)
-        self._add_messages_to_index(new_messages)
+        return self._add_messages_to_index(new_messages)
 
     def size(self):
-        return len(self._messages_seen)
+        return self._message_dao.size()
+    
+    def find(self, message_id : str) -> Message:
+        message = self._message_dao.get(message_id)
+        if message is None:
+            raise ValueError(f'Could not find message: {message_id}')
+        return message
 
-    def _add_messages_to_index(self, new_messages : List[Message]):
+    def _add_messages_to_index(self, new_messages : List[Message]) -> Tuple[Tuple[Message, ...], Tuple[Message, ...]]:
         """ Iterates through all new emails and links together emails that form a thread by populating message.children. """
 
         need_parent : List[Message] = []
+        parent_messages : Set[Message] = set([])
+        patches_to_upload : List[Message] = []
+        patches_with_new_comments : Set[Message] = set([])
 
         # First iterates through all messages to distinguish between 1.) replies and 2.) the start of a thread.
         for message in new_messages:
-            if message.id not in self._messages_seen:
-                self._messages_seen[message.id] = message
+            if not self._message_dao.get(message.id):
+                self._message_dao.store(message)
 
             # If message is a reply, it needs to be associated with an existing thread
             if message.in_reply_to is not None:
                 need_parent.append(message)
                 continue
-
-            # If message doesn't have the LKML style [PATCH v4 00/11],
-            # we don't need to put it in the index.
-            key = message.get_key()
-            if key:
-                self._key_to_thread_start[key] = message
+                
+            parent_messages.add(message)
+            patches_to_upload.append(message)
 
         for message in need_parent:
-            if message.in_reply_to not in self._messages_seen:
+            if not self._message_dao.get(message.in_reply_to):
                 print("Could not find parent email, dropping " + message.subject)
                 continue
-            parent = self._messages_seen[message.in_reply_to]
+            parent = self.find(message.in_reply_to)
             parent.children.append(message)
-
-def find_thread(key: str) -> Message:
-    archive_index = ArchiveMessageIndex()
-    archive_index.update('test_data')
-    return archive_index.find_thread(key)
+            
+            # determines whether the parent of the given email is part of this update or not
+            if parent not in parent_messages:
+                if re.match(r'\[.+ (\d+)/(\d+)\] .+', message.subject):
```

Comment:
this feels like it should be a helper function on Message.

Consider adding such a function in a parent commit.
e.g. 
```python
  def is_patch() -> bool:
     ...
  def patch_index() -> Tuple[int, int]:
    ... # can move parse_set_index from patch_parser.py here too
 
```

Reply:
I can't seem to find resources online to add things in a parent commit. would I basically be rebasing?

Reply:
? Are you asking how to edit a parent commit in a chain of commits? If so, you will probably want to do an interactive rebase ("git rebase -i <parent-commit>^"); you can read more about it here: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_changing_multiple